### ISSUE TITLE
[ECS Provider] - SSM managed instances causes discovery to fail

### ISSUE THREAD

**Issue by JohnPreston** [traefik/traefik#8839]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you do?

Run traefik on an ECS Anywhere ECS instance which the instance ID is not `i-[a-z0-9]+` but `mi-[a-z0-9]+`

I tracked it down as far as 

https://github.com/traefik/traefik/blob/25725e9b2fbd8f383b45c5fb2e3943e1591ab9c1/pkg/provider/ecs/ecs.go#L375
and https://github.com/traefik/traefik/blob/25725e9b2fbd8f383b45c5fb2e3943e1591ab9c1/pkg/provider/ecs/ecs.go#L408

After it found the **ECS Instances** it goes about describing the **EC2 instance** without checking that it is in fact an EC2 instance.

so after that, when trying to build the instance and describe it doesn't work.

The call for ec2-describe-instances is what returns the error. With CLI, for example

```bash
aws ec2 describe-instances --instance-ids mi-08593f2320b5826a3

An error occurred (InvalidInstanceID.Malformed) when calling the DescribeInstances operation: Invalid id: "mi-08593f2320b5826a3"
```

instead we can get some information from SSM

```bash
aws ssm describe-instance-information --instance-information-filter-list key=InstanceIds,valueSet=mi-08593f2320b5826a3
```
See [API Docs](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_DescribeInstanceInformation.html)

One way to catch whether the instance might be external, in addition not having a different instance ID format, is to check for attributes in the **ECS Instance**.

```json
      "attributes": [
        {
          "name": "ecs.capability.external"
        }
]
```

This might not be the best way to detect it, but that can be used in combination of the instance ID format.

I don't know go well enough to open a PR myself for this, sorry about that.

Suggested recommendation: implement to do describe ecs instances and support managed instances
or
ignore that specific error if the instance id does not match the regex `i-[a-z0-9]+`

### What did you see instead?

Results into

```bash

time="2022-03-15T17:38:21Z" level=error msg="Provider connection error failed to get ECS configuration: InvalidInstanceID.Malformed: Invalid id: \"mi-08593f2320b5826a3\"\n\tstatus code: 400, request id: 75eabde6-192c-4406-b5a3-50fcfd816c6e, retrying in 621.405684ms" providerName=ecs

```

### What version of Traefik are you using?

```bash

docker run --rm traefik version
Version:      2.6.1
Codename:     rocamadour
Go version:   go1.17.7
Built:        2022-02-14T16:50:25Z
OS/Arch:      linux/arm64

```

### What is your environment & configuration?

```yaml
---
# Docker compose with overrides and config to run in AWS ECS

version: "3.8"
services:
  traefik:
    ports:
       - 8000:80/tcp
    expose:
      - 8080/tcp
    labels:
      container_name: traefik
    environment:
      AWS_DEFAULT_REGION: ${AWS_DEFAULT_REGION:-eu-west-1}
    command:
      #- "--log.level=DEBUG"
      - "--api.insecure=true"
      - "--providers.ecs.autoDiscoverClusters=true"
      - "--providers.ecs.exposedByDefault=true"
      - "--providers.ecs.refreshSeconds=15"
      - "--providers.ecs.clusters=${CLUSTER_NAME}"
      - "--entrypoints.web.address=:80"
      - "--metrics"
      - '--metrics.prometheus=true'
      - "--metrics.prometheus.addServicesLabels=true"
      - "--metrics.prometheus.addrouterslabels=true"
      - "--entryPoints.web.address=:80"
      - "--entryPoints.web.proxyProtocol.trustedIPs=192.168.103.0/24"
```

It is a modified version of the instructions on https://labs.compose-x.io/apps/traefik_ecs_part1.html which works the same but uses ECS Anywhere instances instead of EC2/Fargate.

### If applicable, please paste the log output in DEBUG level

```bash
time="2022-03-15T17:38:21Z" level=error msg="Provider connection error failed to get ECS configuration: InvalidInstanceID.Malformed: Invalid id: \"mi-08593f2320b5826a3\"\n\tstatus code: 400, request id: 75eabde6-192c-4406-b5a3-50fcfd816c6e, retrying in 621.405684ms" providerName=ecs


```

**Comment by mpl** [comment]
@JohnPreston Did you actually mean to open two issues, or is this one here a duplicate of #8795 as far as you're concerned?

**Comment by JohnPreston** [comment]
Opened a PR #8845
First try at go

**Comment by JohnPreston** [comment]
> @JohnPreston Did you actually mean to open two issues, or is this one here a duplicate of #8795 as far as you're concerned?

Yes I meant to create a different issue.
A. #8795 is more of a statement than anything else IMHO, i.e. "there is no documentation saying it works for ECS Anywhere".
I am very glad that someone pointed it out though. I'd be glad to come up with documentation on how to run in ECS Anywhere but (see below) it should not make any difference given the control plane (ECS) is the same and technically running on on-prem instances is "the same" as in EC2 mode. The only difference would be the no-support of AWSVPC networking mode on prem which from the aspect of figuring out, which IP address to send the traffic to, would be the difference.

B. This issue goes into the why it is not working.

Otherwise, ECS Anywhere discovery as per my use of traefik in ECS so far (and now from attempting to fix by reading the code) does not need to know about the ECS instance (and where applicable, EC2 instance) to allow the services and tasks discovery. Therefore I'd expect that the discovery for these services "as-if" they were running in AWS Fargate, and be able to send the traffic to the tasks of the service based on that discovery.

If however the discovery to send traffic to that instance (on-prem that is) requires to find out the IP address of said instance to send the traffic to it, then my PR needs a big function added to do the discovery of the SSM instance to retrieve the property

**Comment by kevinpollet** [comment]
Closed by #8973
